hibernate-reactive-core/src/main/java/org/hibernate/reactive/adaptor/impl/ResultSetAdaptor.java

```diff
@@ -695,7 +695,7 @@ public Timestamp getTimestamp(int columnIndex, Calendar cal) {
 
 	@Override
 	public Timestamp getTimestamp(String columnLabel, Calendar cal) {
-		throw new UnsupportedOperationException();
+		return getTimestamp(columnLabel);
```

Comment:
Shouldn't you use the `cal` parameter to assign the expected timezone when it's not stored in the db?

Reply:
> Shouldn't you use the `cal` parameter to assign the expected timezone when it's not stored in the db?

Probably, but if so `PreparedStatementAdaptor.setTimestamp()` is broken in the same way.

Reply:
I've just noticed that that in Postgres, if we save an OffsetDateTime we read a different value because of the inconsistency with the timezone. How come we store it as `timestamp` and not `timestampz`?

Reply:
Because postgres doesn't really support `timestamp with timezone`. You can *write* a timestamp with a zone attached, but it just converts it to the local timezone and that's what you get when you read it back.

Very few databases actually support `timestamp with timezone` correctly. It's pretty lame.